powerstore/datasource_volume_group.go

```diff
@@ -0,0 +1,319 @@
+)
+
+type volumeGroupDataSource struct {
+	client *client.Client
+}
+
+type volumeGroupDataSourceModel struct {
+	VolumeGroups []models.VolumeGroupDataSource `tfsdk:"volume_groups"`
+	ID           types.String                   `tfsdk:"id"`
+	Name         types.String                   `tfsdk:"name"`
+}
+
+var (
+	_ datasource.DataSource              = &volumeGroupDataSource{}
+	_ datasource.DataSourceWithConfigure = &volumeGroupDataSource{}
+)
+
+// newVolumeGroupDataSource returns the volume group data source object
+func newVolumeGroupDataSource() datasource.DataSource {
+	return &volumeGroupDataSource{}
+}
+
+func (d *volumeGroupDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_volumegroup"
+}
+
+func (d *volumeGroupDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "VolumeGroup DataSource.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Unique identifier of the volume group.",
+				MarkdownDescription: "Unique identifier of the volume group.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("name")),
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description:         "Volume group name.",
+				MarkdownDescription: "Volume group name.",
+				Optional:            true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"volume_groups": schema.ListNestedAttribute{
+				Description:         "List of volume groups.",
+				MarkdownDescription: "List of volume groups.",
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"id": schema.StringAttribute{
+							Description:         "Unique identifier of the volume group.",
+							MarkdownDescription: "Unique identifier of the volume group.",
+							Computed:            true,
+						},
+						"name": schema.StringAttribute{
+							Description:         "Volume group name.",
+							MarkdownDescription: "Volume group name.",
+							Computed:            true,
+						},
+						"description": schema.StringAttribute{
+							Description:         "Volume group description.",
+							MarkdownDescription: "Volume group description.",
+							Computed:            true,
+						},
+						"creation_timestamp": schema.StringAttribute{
+							Description:         "The time at which the volume group was created.",
+							MarkdownDescription: "The time at which the volume group was created.",
+							Computed:            true,
+						},
+						"is_protectable": schema.BoolAttribute{
+							Description:         "This is a derived field that is set internally.",
```

Comment:
Description and MarkDownDescription is different ?

Reply:
reverted